src/test/java/redis/clients/jedis/tests/SSLJedisClusterTest.java

```diff
@@ -191,11 +191,11 @@ public void connectWithCustomHostNameVerifier() {
       jc2 = new JedisCluster(new HostAndPort("127.0.0.1", 8379), DEFAULT_TIMEOUT, DEFAULT_TIMEOUT,
           DEFAULT_REDIRECTIONS, "cluster", null, DEFAULT_CONFIG, true, 
           null, null, hostnameVerifier, portMap);
-      Assert.fail("The code did not throw the expected NullPointerException.");
-    } catch (NullPointerException e) {
-      // Null pointer exception occurs from closing Jedis object that did not connect due to custom
-      // hostname validation. When closing this Jedis object, the RedisOutputStream in the underlying 
-      // Connection object is null and causes this exception
+      jc2.get("foo");
```

Comment:
@mina-asham `jc2.get("foo");` what is for?

Reply:
It's to trigger the failure, the previous logic used to assign the `socket` variable before failing on hostname verification making the `isConnected` method return `true`, but since now we only assign it if it succeeds the whole thing, it wouldn't behave exactly the same when connecting, so had to trigger the failure some other way.

Reply:
Isn't that a backward compatibility issue? 
@sazzad16

Reply:
For a broken state yes, if you initiate a SSL cluster connection that fails due to host verification you wouldn't get the NPE on connection, but your connection will report broken instead.

Since this is an error state in anyway, I think the backward compatibility issue is not bad, what do you think?

Reply:
@gkorland Kinda! Only if a user is catching `NullPointerException` without catching `JedisConnectionException`/`JedisException`. IMHO, this could be very rare and shouldn't be a significant issue.